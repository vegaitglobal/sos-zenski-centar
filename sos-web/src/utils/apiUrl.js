export const baseUrl = process.env.REACT_APP_BASE_URL;
//CREATE .ENV IN SOS-WEB AND DEFINE REACT_APP_BASE_URL ('http://localhost:48201') 
