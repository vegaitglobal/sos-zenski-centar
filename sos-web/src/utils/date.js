const formatDate = (date) => {
  let validDate = date.toLocaleDateString('en-GB');
  const valid = (() => {
    const day = validDate.split('/')[0];
    const month = validDate.split('/')[1];
    const year = validDate.split('/')[2];
    return `${year}-${month}-${day}`;
  })();
  return valid;
};

export const getLastMonth = () => {
  const today = new Date();

  const firstDay = new Date(today.getFullYear(), today.getMonth() - 1, 1);
  const lastDay = new Date(today.getFullYear(), today.getMonth(), 0);

  return {
    firstDay: formatDate(firstDay),
    lastDay: formatDate(lastDay),
  };
};
