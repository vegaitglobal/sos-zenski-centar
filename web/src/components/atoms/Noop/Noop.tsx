import { ReactElement } from 'react';

export const Noop = (): ReactElement => <></>;
