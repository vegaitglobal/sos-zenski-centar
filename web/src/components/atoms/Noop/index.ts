export { Noop as default } from './Noop';
