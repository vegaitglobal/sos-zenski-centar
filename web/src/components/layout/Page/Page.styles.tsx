import styled from 'styled-components';

export const StyledPage = styled.div`
	padding: 10vh 0;
`;
