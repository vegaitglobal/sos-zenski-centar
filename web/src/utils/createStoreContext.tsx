import { createContext, PropsWithChildren, FunctionComponent, Context, useContext } from 'react';
import useStoreFactory from '../hooks/useStoreFactory';

/**
 * Creates a context for the store created by the passed `storeFactory`.
 * Additionally returns the `Provider` component and the hook for the created store's context.
 *
 * @param name - name of the store
 * @param storeFactory - a function that returns an object representing the store
 */
// eslint-disable-next-line @typescript-eslint/naming-convention
const createStoreContext = <TStore extends Record<string, unknown>>(
	name: string,
	storeFactory: () => TStore,
): [FunctionComponent<unknown>, () => TStore, Context<TStore>] => {
	const context = createContext<TStore>({} as TStore); // the initial value is only to satisfy TS and won't be used

	/**
	 * Initializes and contains the context for the `TStore`.
	 */
	function StoreProvider({ children }: PropsWithChildren<unknown>): JSX.Element {
		const store = useStoreFactory(storeFactory);
		return <context.Provider value={store}>{children}</context.Provider>;
	}

	const useStore = (): TStore => {
		const store = useContext(context);
		if (store) {
			return store;
		}
		throw new ReferenceError(
			`${name} is not initialized, make sure that your component is wrapped in the 'Provider' of the store.`,
		);
	};

	return [StoreProvider, useStore, context];
};

export default createStoreContext;
