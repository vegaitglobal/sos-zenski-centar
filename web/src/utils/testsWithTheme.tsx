import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { ReactElement, ComponentType, ReactNode } from 'react';
import { theme } from '../styles/config/theme';

export const componentWithTheme = (component: ReactNode): ReactElement => (
	<ThemeProvider theme={theme}>{component}</ThemeProvider>
);

export const renderWithTheme = (component: ReactElement, options?: RenderOptions): RenderResult => {
	const wrapper: ComponentType = ({ children }) => componentWithTheme(children);

	return render(component, { wrapper, ...options });
};
