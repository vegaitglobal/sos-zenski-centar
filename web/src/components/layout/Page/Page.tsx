import { ReactElement } from 'react';
import { PageProps } from './Page.data';
import { StyledPage } from './Page.styles';

export const Page = ({ children, ...props }: PageProps): ReactElement => (
	<StyledPage data-testid="Page" {...props}>
		{children}
	</StyledPage>
);
