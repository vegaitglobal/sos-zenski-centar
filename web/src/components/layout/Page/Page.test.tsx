import renderer from 'react-test-renderer';
import Noop from '../../atoms/Noop';
import Page from './index';
import { renderWithTheme } from '../../../utils/testsWithTheme';

describe('Page', () => {
	describe('Render', () => {
		it('Renders Page', () => {
			const { getByTestId } = renderWithTheme(
				<Page>
					<Noop />
				</Page>,
			);

			const element = getByTestId('Page');

			expect(element).toBeTruthy();
		});

		it('Renders correctly', () => {
			const tree = renderer
				.create(
					<Page>
						<Noop />
					</Page>,
				)
				.toJSON();

			expect(tree).toMatchSnapshot();
		});
	});

	describe('Content', () => {
		it('Can have elements as children', () => {
			const { getByTestId } = renderWithTheme(
				<Page>
					<span>Child</span>
				</Page>,
			);

			const element = getByTestId('Page');
			const child = element.querySelector('span');

			expect(element).toContainElement(child);
		});
	});
});
