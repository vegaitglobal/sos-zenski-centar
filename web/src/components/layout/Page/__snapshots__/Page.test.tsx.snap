// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Render Renders correctly 1`] = `
<div
  className="sc-bdnxRM bQBUVB"
  data-testid="Page"
/>
`;
